src/Microsoft.AspNetCore.WebUtilities/FileBufferingReadStream.cs

```diff
@@ -135,22 +180,27 @@ private Stream CreateTempFile()
         public override int Read(byte[] buffer, int offset, int count)
         {
             ThrowIfDisposed();
-            if (_buffer.Position < _buffer.Length || _completelyBuffered)
+            if (_buffer.Position < _readIntoBuffer || _completelyBuffered)
```

Comment:
Bleh, sorry has been a while since this PR :)

Using a fixed sized memory stream rather than a resizable one; length in the fixed size case is the size of the underlying array not its used space (different behavior between the two inits :-/)